### PR TITLE
support xml in generated services

### PR DESCRIPTION
**Pull request by ctaggart**
Fixes :
- #649
- #1046

It updates the model to support xml serialization of a wrapped array. It adds support for the `x-ms-version` header. The `Cargo.toml` is now generated using a template and will use the `azure_core` `xml` feature when needed.

All features have been checked:
Check all is here: https://github.com/ctaggart/azure-sdk-for-rust/actions/runs/3005186676

When reviewing, if you deselect *.toml files, it is easier.

### CODE REVIEW

File: services/svc/blobstorage/examples/list_containers.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    let endpoint = format!("https://{}.blob.core.windows.net", account_name);
+    let scopes = &["https://storage.azure.com/"];
+    let credential = Arc::new(AzureCliCredential::new());
+    let client = azure_svc_blobstorage::Client::builder(credential)
+        .endpoint(endpoint)
+        .scopes(scopes)
+        .build();
+
+    let x_ms_version = "2021-08-06";
```
Comment (cataggar) [inline services/svc/blobstorage/examples/list_containers.rs:25]: Shouldn't have to set this.

File: services/svc/blobstorage/src/package_2021_08/models.rs [hunk 2]
```diff
@@ -1185,25 +1185,30 @@ pub struct ListContainersSegmentResponse {
...
-    #[serde(rename = "ContainerItems")]
-    pub container_items: Vec<ContainerItem>,
+    #[serde(rename = "Containers")]
+    pub containers: Containers,
     #[serde(rename = "NextMarker", default, skip_serializing_if = "Option::is_none")]
     pub next_marker: Option<String>,
 }
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, Default)]
+pub struct Containers {
+    #[serde(rename = "$value", default, skip_serializing_if = "Vec::is_empty")]
```
Comment (cataggar) [inline services/svc/blobstorage/src/package_2021_08/models.rs:1195]: This is the xml wrapper object.

File: services/svc/blobstorage/src/package_2021_08/models.rs [hunk 3]
```diff
@@ -1185,25 +1185,30 @@ pub struct ListContainersSegmentResponse {
...
 impl ListContainersSegmentResponse {
-    pub fn new(service_endpoint: String, container_items: Vec<ContainerItem>) -> Self {
+    pub fn new(service_endpoint: String) -> Self {
         Self {
             service_endpoint,
             prefix: None,
             marker: None,
             max_results: None,
-            container_items,
+            containers: Default::default(),
```
Comment (cataggar) [inline services/svc/blobstorage/src/package_2021_08/models.rs:1211]: It is a wrapper object, just default it.

File: services/autorust/codegen/src/codegen_models.rs [hunk 4]
```diff
@@ -50,7 +59,45 @@ impl SchemaGen {
...
+    /// <books><book/><book/></books>) or unwrapped (<book/><book/>). Default value is false.
+    /// The definition takes effect only when defined alongside type being array (outside the items).
+    ///
+    /// https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md#xmlObject
+    fn xml_wrapped(&self) -> bool {
+        self.schema.common.xml.as_ref().map(|xml| xml.wrapped).flatten().unwrap_or_default()
+    }
+
     fn name(&self) -> Result<&str> {
+        // if self.xml_wrapped() {
```
Comment (cataggar) [inline services/autorust/codegen/src/codegen_models.rs:96]: uncomment

File: services/svc/blobstorage/examples/list_containers.rs [hunk 5]
```diff
@@ -0,0 +1,38 @@
+/*
+List blob containers in a storage account
+cargo run --package azure_svc_blobstorage --example list_containers $STORAGE_ACCOUNT_NAME
```
Comment (ctaggart) [inline services/svc/blobstorage/examples/list_containers.rs:3]: This is a new example.

File: services/autorust/codegen/templates/Cargo.toml.jinja [hunk 6]
```diff
@@ -1,6 +1,38 @@
...
+description = "generated REST API bindings"
+repository = "https://github.com/azure/azure-sdk-for-rust"
+homepage = "https://github.com/azure/azure-sdk-for-rust"
+documentation = "https://docs.rs/{{package_name}}"
+
+[lib]
+doctest = false
+
+[dependencies]
+azure_core = { path = "../../../sdk/core", version = "0.4", features = [{%- for feature in azure_core_features -%}"{{feature}}", {%- endfor -%}] }
```
Comment (bmc-msft) [inline services/autorust/codegen/templates/Cargo.toml.jinja:16]:
```suggestion
azure_core = { path = "../../../sdk/core", version = "0.4"
{%- let len = azure_core_features.len() -%}
{%- if len != 0 -%}
,features = [
{%- for feature in azure_core_features -%}"{{feature}}" {%- if !loop.last -%}, {%- endif -%} {%- endfor -%}
]
{%- endif -%}
}
```

**Review by bmc-msft — APPROVED** [review]

File: services/autorust/codegen/src/content_type.rs [hunk 7]
```diff
@@ -3,11 +3,17 @@ pub const APPLICATION_XML: &str = "application/xml";
...
+    if value == Some(APPLICATION_JSON) {
+        return value;
     }
-    consumes.into_iter().next()
+    for next in list {
+        if next == APPLICATION_JSON {
+            return Some(next);
+        }
+    }
+    value
```
Comment (rylev) [inline services/autorust/codegen/src/content_type.rs:16]: Can this be written like this:

```rust
pub fn pick2<'a>(mut list: impl Iterator<Item = &'a str> + Clone) -> Option<&'a str> {
    list.clone().find(|&value| value == APPLICATION_JSON).or_else(|| list.next())
}
```

**Pull request merged**